debian/rules

```diff
@@ -1,8 +1,14 @@
 #!/usr/bin/make -f
 
-QT_SELECT=qt5
-export QT_SELECT
+export DEB_BUILD_MAINT_OPTIONS = hardening=+all
```

Comment:
I thought those flags would be passed to the build via environment variables. Isn't it best practice to rely on env variables passed by e.g. launchpad.net's build system instead of hard-coding them ourselves?

(I've never fully read the debian maintainer guide, so no idea what's best practice :) )

Reply:
The package maintainer should know best which flags are safe to use for the package. So the build environment doesn't impose much on the package.
Make doesn't mind the spaces and this is taken directly from the [Guide for Debian Maintainers](https://www.debian.org/doc/manuals/debmake-doc/ch08.en.html#cmake-single)
The _Debian New Maintainers' Guide_ is replaced by the _Guide for Debian Maintainers_. The new guide is much better, and has more and better examples.

Reply:
Great, thanks! All those flags are what I expected launchpad.net to add anyway.

I think we could even switch to Qt 6. I will use the "old" `debian/` files for Ubuntu 20 and later anyway.
But you don't need to do that! I want to update all build-instructions to Qt6 next year anyway and will try to update this then.

I also thought about moving these files to a separate "packaging" repository once. But I'm still unsure whether that's a good idea. 😄

Reply:
In my company we use http://honk.sigxcpu.org/projects/git-buildpackage/manual-html/gbp.html and different branches for the different distributions.
Then the debianization is located only on the distribution branch. This mirrors the way Debian/Ubuntu are managing packages. As they use the software from upstream and put a debianization on top of it.